### PR TITLE
Windows compatibility fixes for bytecode report

### PR DESCRIPTION
**Pull request by cameel**
- Both `isolate_tests.py` and `prepare_report.py` were using Python's universal newline mode when reading/writing files (by using the default `newline=None` with [open()](https://docs.python.org/3/library/functions.html#open)). This resulted in files stored on disk having `\n` newlines on Unix and `\r\n` on Windows. It didn't matter with Standard JSON because the universal newline mode converts them all to `\n` when reading the file and inserting it into JSON but with the new upcoming CLI mode (#10676) the file is read from disk and bytecode ends up different on Windows.
- My updated `prepare_report.py` from #10675 would print Windows-style paths on Windows. This only affected unit tests since in practice we only pass file names without path to the report.
- Turns out that docstrings get different line endings on different platforms. I was using them in unit tests added in #10675. This PR converts them all to normal strings with explicit `\n` endings.
- Git in CircleCI is configured with `core.autocrlf = true` on Windows machines which means that the checked out files are not exactly identical to the ones in the repo. I modified Windows jobs to disable the conversion.

Note that don't have a CI job for running Python unit tests yet. I'm adding one in #10819 and basing it on top of this PR so you can see there if they're passing.

### CODE REVIEW

**Comment by cameel** [comment]
Bytecode comparison fails so it looks like I'm not done with this PR yet.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `894af67e8e90799f47053b52abba47ce31e5434c`:
```
Error: Trailing whitespace found:
 test/scripts/fixtures/smt_contract_with_cr_newlines.sol:1:pragma experimental SMTChecker;^M^Mcontract C {^M}^M
 test/scripts/fixtures/smt_contract_with_crlf_newlines.sol:1:pragma experimental SMTChecker;^M
 test/scripts/fixtures/smt_contract_with_crlf_newlines.sol:2:^M
 test/scripts/fixtures/smt_contract_with_crlf_newlines.sol:3:contract C {^M
 test/scripts/fixtures/smt_contract_with_crlf_newlines.sol:4:}^M
 test/scripts/fixtures/smt_contract_with_mixed_newlines.sol:3:contract C {^M}^M

```
Please check that your changes are working as intended.

**Comment by cameel** [comment]
Turns out that @chriseth was right, git does automatic line ending conversion on Windows in CircleCI. I researched that some time ago and figured out it does not because the official docs do not specify the default and found info in other sources that it's `false`. Turns out that it is but the installer on Windows asks the user about it (so it depends) and CircleCI and Github decided to enable it too (see [CircleCI config teardown: How we write our CircleCI config at CircleCI](https://circleci.com/blog/circleci-config-teardown-how-we-write-our-circleci-config-at-circleci/) and https://github.com/actions/checkout/issues/135#issuecomment-613329879).

I checked via SSH and indeed it's the input file checked out from the repo contains `\r\n` endings and it's not Python's `open()` changing them.

So I think this time I got all of it and bytecode comparison should now be passing but we'll see...

File: test/scripts/test_bytecodecompare_prepare_report.py
Comment (hrkrshnn) [inline test/scripts/test_bytecodecompare_prepare_report.py]: This can be removed now, right?
```suggestion
```

**Review by hrkrshnn — COMMENTED** [review]
Looks good. Minor comment

File: test/scripts/test_bytecodecompare_prepare_report.py
Comment (cameel) [reply test/scripts/test_bytecodecompare_prepare_report.py]: Right. Good catch. I wonder why it passed our pylint checks. Looks like I need to take a look at `pylintrc`.

Comment (cameel) [reply test/scripts/test_bytecodecompare_prepare_report.py]: Fixed.

As for pylint, turns out the warning is disabled in the configuration :) Most of the stuff that's disabled is really easy to fix so I'm preparing a PR to re-enable it.

**Review by hrkrshnn — DISMISSED** [review]
LGTM. One more review would be nice before merging.

**Comment by leonardoalt** [comment]
@cameel conflicts

**Comment by cameel** [comment]
ok. Rebasing now.

**Review by leonardoalt — DISMISSED** [review]
lgtm as well when the conflicts are fixed

**Comment by cameel** [comment]
Done. The only conflict was in `check_style.sh`, due to changed indentation of the list of files exempt from trailing whitespace check.

**Review by leonardoalt — APPROVED** [review]

**Review by leonardoalt — APPROVED** [review]

**Review by hrkrshnn — APPROVED** [review]

**Pull request merged**
